### PR TITLE
Sleep repo branch

### PR DESCRIPTION
**Pull request by spaceplesiosaur**
This branch is broken, but I am not sure why and I'm hoping someone can help me out.  I made changes to SleepRepo.js to refactor it, but for some reason the tests aren't passing as expected.  Here is a screenshot of my failure:

Screen Shot 2019-10-20 at 10 34 58 AM
 
I'll keep looking at it, but if someone can see what I'm missing, lmk.  The original code is still in the file, just commented out.

### CODE REVIEW

**Comment by spaceplesiosaur** [comment]
@brittanystoroz @khalidwilliams This branch has code that isn't working properly that I'm hoping my teammates will look at since I've gotten stuck.  Is this the right way to go about this?  I put the clean code in a different branch with a different PR.

**Comment by spaceplesiosaur** [comment]
Figured out what's happening, don't know why it's happening.  Even though I used the exact same code, the array of user data is not splicing out into a week - it's just staying the same, so data that isn't supposed to be there is there, and affecting the numbers.  Don't know why.

**Comment by brittanystoroz** [comment]
There are a lotttt of changes in this PR that are making it really difficult to read and help out. You have changes in Activity, changes in your jQuery, etc. When submitting a PR for review, you want to ONLY include the bare minimum amount of changes that you are trying to discuss/get feedback on. I would recommend looking into `git add --patch` that will allow you to stage separate chunks of changes for commit/PR rather than adding every change you've made. 

If you can open a new PR with just the relevant changes, it will be much easier for us to jump in and help identify what was going wrong!

File: src/SleepRepo.js [hunk 1]
```diff
@@ -9,21 +9,43 @@ class SleepRepo {
...
+    //   }
+    //   return arr;
+    // }, []);
+
+    // let avgSleepQualityPerUser = dataByUser.map(user => [...user].splice(-7 * week, 7)).map(user => user.reduce((totalQuality, day) => {
+    //   totalQuality += day.sleepQuality;
+    //   return totalQuality;
+    // }, 0)).map(user => Number((user / 7).toFixed(2)));
+
+    let avgSleepQualityPerUser = this.returnAvgSleepDataPerUser(dataByUser, 'sleepQuality').map(user => Number((user / 7).toFixed(2)));
```
Comment (ckoga) [inline src/SleepRepo.js]: The first argument of returnAvgSleepDataPerUser dataByUser is supposed to be a number.  I think you use week in other methods.

**Comment by spaceplesiosaur** [comment]
Fixed the variable!

File: src/SleepRepo.js [hunk 2]
```diff
@@ -40,24 +49,15 @@ class SleepRepo {
...
-        arr[user.userID - 1] = [user];
-      } else {
-        arr[user.userID - 1].push(user);
-      }
-      return arr;
-    }, []);
+  returnWeeklyLongestSleepers(week, relevantData) {
+    let dataByUser = this.returnDataByUser();
+
+    let avgSleepHoursPerUser = this.returnAvgSleepDataPerUser(week, 'hoursSlept')
```
Comment (ckoga) [inline src/SleepRepo.js]: Sweet everything looks good.  The tests are all passing go ahead and merge it!

**Pull request merged**